### PR TITLE
Update to Git SHA instead of SVN revision number.

### PR DESCRIPTION
**Pull request by pshriwise**
## Description
I noticed today that the value `"$Rev"` was present in the configured and installed `DagMCVersion.hpp` file, which didn't seem right. We were last using SVN in ~2010.

This PR updates commit revision tracking from an SVN revision number to a Git SHA. I've deprecated the `DagMC::interface_revision` method and added a `DagMC::git_sha` method. I'm fine with simply removing the `interface_revision` function, but it seemed proper to deprecate it first and since I doubt anyone is using it I don't think it's doing any harm either way

### CODE REVIEW

**Review by bam241 — APPROVED** [review]
LGTM!

File: src/dagmc/DagMC.cpp [hunk 1]
```diff
@@ -112,28 +112,7 @@ float DagMC::version(std::string* version_string) {
...
-    if (endptr == interface_string.c_str() + key_str.size()  // parsing end
-        || ((result == LONG_MAX || result == LONG_MIN)
-            && errno == ERANGE)) {  // checking range
-      std::cerr << "Not able to parse revision number" << std::endl;
-      exit(1);
-    }
-    return result;
-  }
-  return result;
+  return 0;
```
Comment (bam241) [inline src/dagmc/DagMC.cpp]: maybe move this in the header ?

**Review by gonuke — APPROVED** [review]
LGTM!

**Comment by bam241** [comment]
thx @pshriwise ! merging!

**Pull request merged**